### PR TITLE
Issue #7604: Update doc NonEmptyAtclauseDescriptionCheck

### PR DESCRIPTION
**Pull request by MauryanKansara**
Issue #7604 

Screenshot (12)

`$ cat config1.xml`
```cmd
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
  <module name="TreeWalker">
    <module name="NonEmptyAtclauseDescription"></module>
  </module>
</module>

```
`$ cat Test1.java`
```java
class Test1
{
  /**
  * @param a Multiplying a with 10.
  * @param ans
  * @deprecated
  * @throws Exception
  */
  public int method(int a, int ans) throws Exception
  {
    ans = a * 10;
    return ans;
  }
}
```

`$ java -jar checkstyle-10.6.0-all.jar -c config1.xml Test1.java`
```cmd
Starting audit...
[ERROR] C:\CheckStyle\issues\Test1.java:5: At-clause should have a non-empty description. [NonEmptyAtclauseDescription]
[ERROR] C:\CheckStyle\issues\Test1.java:6: At-clause should have a non-empty description. [NonEmptyAtclauseDescription]
[ERROR] C:\CheckStyle\issues\Test1.java:7: At-clause should have a non-empty description. [NonEmptyAtclauseDescription]
Audit done.
Checkstyle ends with 3 errors.
```
Screenshot (13)

`$ cat config2.xml`
```cmd
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
  <module name="TreeWalker">
    <module name="NonEmptyAtclauseDescription">
        <property name="javadocTokens" value="PARAM_LITERAL,RETURN_LITERAL"/>
    </module>
  </module>
</module>
```
`$ cat Test2.java`
```java
class Test
{
  // OK for parameter "a" because it's description is provided.
  // Violation for parameter "ans" because it's description is not provided.
  // "deprecated" and "throws" tags are ignored.
  /**
  * @param a Multiplying a with 10.
  * @param ans
  * @deprecated
  * @throws Exception
  */
  public int method(int a, int ans) throws Exception
  {
    ans = a * 10;
    return ans;
  }
}
```
`$ java -jar checkstyle-10.6.0-all.jar -c config2.xml Test2.java`
```cmd
Starting audit...
[ERROR] C:\CheckStyle\issues\Test2.java:8: At-clause should have a non-empty description. [NonEmptyAtclauseDescription]
Audit done.
Checkstyle ends with 1 errors.
```

Screenshot (14)

### CODE REVIEW

**Comment by MauryanKansara** [comment]
@romani can you please review my PR?

**Comment by nrmancuso** [comment]
Github, generate site

**Review by romani — CHANGES_REQUESTED** [review]
items:

File: config/archunit-store/slices_should_be_free_of_cycles_suppressions [hunk 1]
```diff
@@ -768,13 +768,13 @@ Cycle detected: Slice api -> \
     - Method <com.puppycrawl.tools.checkstyle.checks.javadoc.JavadocTagInfo$6.isValidOn(com.puppycrawl.tools.checkstyle.api.DetailAST)> gets field <com.puppycrawl.tools.checkstyle.api.Scope.PRIVATE> in (JavadocTagInfo.java:150)\
     - Method <com.puppycrawl.tools.checkstyle.checks.javadoc.JavadocMissingLeadingAsteriskCheck.visitJavadocToken(com.puppycrawl.tools.checkstyle.api.DetailNode)> calls method <com.puppycrawl.tools.checkstyle.api.DetailNode.getType()> in (JavadocMissingLeadingAsteriskCheck.java:151)\
     - Constructor <com.puppycrawl.tools.checkstyle.checks.javadoc.JavadocContentLocationCheck.<init>()> calls constructor <com.puppycrawl.tools.checkstyle.api.AbstractCheck.<init>()> in (JavadocContentLocationCheck.java:154)\
-    - Method <com.puppycrawl.tools.checkstyle.checks.javadoc.NonEmptyAtclauseDescriptionCheck.visitJavadocToken(com.puppycrawl.tools.checkstyle.api.DetailNode)> calls method <com.puppycrawl.tools.checkstyle.api.DetailNode.getParent()> in (NonEmptyAtclauseDescriptionCheck.java:155)\
```
[the remark this answers is not in the document]
Comment (romani) [reply config/archunit-store/slices_should_be_free_of_cycles_suppressions:771]: if diff is in line number update, due to extra javadoc, it is ok to keep

**Comment by romani** [comment]
please make example like:
```
<module name="NonEmptyAtclauseDescription"/>

class Test
{
  // '@param a' - OK
  // '@param ans' - OK
  // '@return' - OK
  // '@deprecated' - OK
  // '@throws Exception' - OK
  /**
  * @param a Multiplying a with 10.
  * @param ans our accumulation parameter
  * @deprecated As of release 1.3
  * @return multiplied value
  * @throws Exception if something is bad
  */
  public int method(int a, int ans) throws Exception
  {
    ans = a * 10;
    return ans;
  }
}

<module name="NonEmptyAtclauseDescription"/>

class Test
{
  // '@param a' - OK for parameter "a" because it's description is provided.
  // '@param ans' - violation
  // '@return' - violation
  // '@deprecated' - violation
  // '@throws Exception' - violation
  /**
  * @param a Multiplying a with 10.
  * @param ans
  * @deprecated
  * @return
  * @throws Exception
  */
  public int method(int a, int ans) throws Exception
  {
    ans = a * 10;
    return ans;
  }
}

<module name="NonEmptyAtclauseDescription">
  <property name="javadocTokens" value="PARAM_LITERAL,RETURN_LITERAL"/>
</module>
        
class Test
{
  // '@param a' - OK because it's description is provided.
  // '@param ans' - violation
  // '@return' - violation
  // '@deprecated' - ok because not a target of validation
  // '@throws Exception' - ok because not a target of validation
  /**
  * @param a Multiplying a with 10.
  * @param ans
  * @deprecated
  * @return
  * @throws Exception
  */
  public int method(int a, int ans) throws Exception
  {
    ans = a * 10;
    return ans;
  }
}
```

**Comment by romani** [comment]
@MauryanKansara , please finish this PR.
if you have no time now, we can let others to finish it.

**Comment by MauryanKansara** [comment]
> @MauryanKansara , please finish this PR. if you have no time now, we can let others to finish it.

I will finish the pending work within a day. Really sorry for the delay.

**Comment by romani** [comment]
we lost contact with author.

**Pull request closed without merging**